src/emqx_client.erl

```diff
@@ -151,6 +155,23 @@
 %%------------------------------------------------------------------------------
 %% API
 %%------------------------------------------------------------------------------
+-spec(def_response(client(), function() | binary()) -> ok).
+def_response(Responser, Response) ->
+    gen_statem:call(Responser, {def_response, Response}).
+
+sub_response_topic(Client, Shared, RequestQoS, Topic) ->
+    Properties = gen_statem:call(Client, request_info),
+    case maps:find('Response-Information', Properties) of
+        {ok, ResponseInformation} ->
+            NewResponseTopic = <<(shared_topic(Shared, ResponseInformation))/binary,
+                                 "/",
+                                 (emqx_base62:encode(Topic))/binary>>,
+            {ok, _Props, _QoS} = subscribe(Client, [{NewResponseTopic, [{rh, 2}, {rap, false},
```

Comment:
It is perhaps helpful to add a log (debug?) to print Props and QoS for troubleshooting